src/components/FidelityBondDevOnly.tsx

```diff
@@ -0,0 +1,350 @@
+import React, { useEffect, useMemo, useState } from 'react'
+import * as rb from 'react-bootstrap'
+import { Trans, useTranslation } from 'react-i18next'
+import { Link } from 'react-router-dom'
+import { useCurrentWallet, useReloadCurrentWalletInfo, Utxos } from '../context/WalletContext'
+import { CopyButtonWithConfirmation } from '../components/CopyButton'
+import { isFeatureEnabled } from '../constants/features'
+
+// @ts-ignore
+import PageTitle from './PageTitle'
+// @ts-ignore
+import DisplayUTXOs from './DisplayUTXOs'
+
+import { routes } from '../constants/routes'
+import * as Api from '../libs/JmWalletApi'
+import styles from './FidelityBond.module.css'
+
+type AlertWithMessage = rb.AlertProps & { message: string }
+
+const dateToLockdate = (date: Date): Api.Lockdate =>
+  `${date.getUTCFullYear()}-${date.getUTCMonth() >= 9 ? '' : '0'}${1 + date.getUTCMonth()}` as Api.Lockdate
+
+const lockdateToTimestamp = (lockdate: Api.Lockdate): number => {
+  const split = lockdate.split('-')
+  return Date.UTC(parseInt(split[0], 10), parseInt(split[1], 10) - 1, 1, 0, 0, 0)
+}
+
+// a maximum of years for a timelock to be accepted
+// this is useful in simple mode - when it should be prevented that users
+// lock up their coins for an awful amount of time by accident.
+// in "advanced" mode, this can be dropped or increased substantially
+const DEFAULT_MAX_TIMELOCK_YEARS = 10
+
+const initialLockdate = () => {
+  const now = new Date()
+  const year = now.getUTCFullYear()
+  const month = now.getUTCMonth()
+  return new Date(Date.UTC(year + 1, month, 1, 0, 0, 0))
+}
+
+interface LockdateFormProps {
+  onChange: (lockdate: Api.Lockdate) => void
+  maxYears?: number
+}
+const LockdateForm = ({ onChange, maxYears = DEFAULT_MAX_TIMELOCK_YEARS }: LockdateFormProps) => {
+  const { t } = useTranslation()
+
+  const now = new Date()
+  const currentYear = now.getUTCFullYear()
+  const currentMonth = 1 + now.getUTCMonth() // utc month ranges from [0, 11]
+
+  const [lockdateYear, setLockdateYear] = useState(currentYear + 1)
+  const [lockdateMonth, setLockdateMonth] = useState(currentMonth)
+
+  useEffect(() => {
+    const date = new Date(Date.UTC(lockdateYear, lockdateMonth - 1, 1, 0, 0, 0))
+    console.log(date.toLocaleDateString())
+    onChange(dateToLockdate(date))
+  }, [lockdateYear, lockdateMonth, onChange])
+
+  const minMonth = () => {
+    if (lockdateYear > currentYear) {
+      return 1
+    }
+
+    return currentMonth + 1 // can be '13' - which means it never is valid and user must adapt 'year'.
+  }
+
+  return (
+    <rb.Row>
+      <rb.Col xs={6}>
+        <rb.Form.Group className="mb-4" controlId="locktimeYear">
```

Comment:
It's a shame that Firefox still [doesn't support](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/month#browser_compatibility) the "month" date picker. It's been tracked for over 9 years... Same for Safari.

Anyway what do you think of a dropdown here? At least for the months a 12 value dropdown is something users will know from most websites. The same applies to the year as well I guess. Most users won't pick a date several decades ahead (I guess) so the year will be accessible quickly in the dropdown.

Longer term, we should go with a nice date picker of course but this can be an improvement. Shame on you Firefox and Safari.

Reply:
can we implement something like this? 
<img width="405" alt="image" src="https://user-images.githubusercontent.com/96262473/172227144-651e9668-4ab4-4c4f-9557-6f46ccc06dfc.png">
Source: https://www.primefaces.org/primeng/calendar

Reply:
found this, maybe it helps: https://raygun.com/blog/building-great-looking-useable-date-time-picker-without-bootstrap/